File: src/test/fixtures/kotlin/LogicalControlFlow.kt
// clean — not flagged by logical control flow
fun cleanExplicitIf(isLoggedIn: Boolean) {
    if (isLoggedIn) {
        navToDashboard()
    }
}

// clean — logical control flow shorthand (cond && action / cond || action) not detected in Kotlin; grammar limitation
fun flaggedAndAsIf(isLoggedIn: Boolean) {
    isLoggedIn && navToDashboard()
}

// clean — logical control flow shorthand (cond && action / cond || action) not detected in Kotlin; grammar limitation
fun flaggedOrAsUnless(isLoggedIn: Boolean) {
    isLoggedIn || navToDashboard()
}

File: src/test/fixtures/python/logical_control_flow.py
# clean — not flagged by logical control flow
def cleanExplicitIf(is_logged_in):
    if is_logged_in:
        nav_to_dashboard()


# flagged — logical control flow (low)
def flaggedAndAsIf(is_logged_in):
    is_logged_in and nav_to_dashboard()


# flagged — logical control flow (low)
def flaggedOrAsUnless(is_logged_in):
    is_logged_in or nav_to_dashboard()
